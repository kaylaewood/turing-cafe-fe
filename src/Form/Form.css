.reserve-button {
  background-color: teal;
  color: white;
}

input,
.reserve-button,
.sort-button {
  margin-right: 2vw;
  font-size: 1.5vw;
  padding: .5vw;
  width: 15vw;
  border: none;
  border-radius: 5px;
}

.sort-button {
  background-color: #FFBF00;
  float: left;
  margin-top: 1vw;
}
