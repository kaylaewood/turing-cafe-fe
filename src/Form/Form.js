import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  render() {
    return (
      <form>
        <input
          type='text'
          name='name'
          value={this.state.name}
          placeholder='Name'
          onChange={event => this.handleChange(event)}
        />
        <input
          type='text'
          name='date'
          value={this.state.date}
          placeholder='Date (mm/dd)'
          onChange={event => this.handleChange(event)}
        />
        <input
          type='text'
          name='time'
          value={this.state.time}
          placeholder='Time'
          onChange={event => this.handleChange(event)}
        />
        <input
          type='text'
          name='number'
          value={this.state.number}
          placeholder='Number of Guests'
          onChange={event => this.handleChange(event)}
        />
        <button className='reserve-button' onClick={() => this.props.submitReservation(this.state.name, this.state.date, this.state.time, this.state.number)}>Make Reservation</button>
      </form>
    )
  }
}

export default Form;
