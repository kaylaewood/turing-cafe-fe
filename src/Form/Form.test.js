import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let submitReservationMock = jest.fn();
  let wrapper;

  beforeEach(() => {
      wrapper = shallow(<Form
      submitReservation={submitReservationMock}
      />)
  })

  it('should match snapshot as expected', () => {

    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when handleChange is invoked', () => {
    const event = {
      target: {
        name: 'name',
        value: 'Robbie'
      }
    }

    wrapper.instance().handleChange(event);
    expect(wrapper.state('name')).toEqual('Robbie');
  })

  it('should invoke submitReservation when button is clicked with correct information', () => {
    wrapper.setState({
      name: 'Eric',
      date: '12/03',
      time: '1:30',
      number: '4'
    })
    wrapper.find('button').simulate('click');
    expect(submitReservationMock).toHaveBeenCalledWith('Eric', '12/03', '1:30', '4');
  })
})
