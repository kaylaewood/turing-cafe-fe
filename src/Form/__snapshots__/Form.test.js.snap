// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot as expected 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    className="reserve-button"
    onClick={[Function]}
  >
    Make Reservation
  </button>
  <br />
  <button
    className="sort-button"
    onClick={[Function]}
  >
    Sort by Date
  </button>
</form>
`;
